src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Security/InputValidator.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            "&&", "||", "&", "|", ";"
+        };
+
+        private static ILogger? Logger;
+
+        /// <summary>
+        /// Sets the logger to be used by all validation methods.
+        /// This should be called once during application startup.
+        /// </summary>
+        public static void SetLogger(ILogger logger)
+        {
+            ArgumentNullException.ThrowIfNull(logger);
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Validates a path for potential security issues like path traversal attacks.
+        /// </summary>
+        /// <param name="path">The path to validate</param>
+        /// <param name="pathType">Description of the path type for logging</param>
+        /// <returns>ValidationResult indicating if path is safe</returns>
+        public static ValidationResult ValidateDirTraversal(string? path, string pathType = "path")
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return ValidationResult.Invalid($"{pathType} cannot be null or empty");
+            }
+
+            return ValidationResult.Valid(path);
+        }
+
+        /// <summary>
+        /// Validates a TypeSpec path for security and format.
+        /// </summary>
+        /// <param name="path">The TypeSpec path to validate</param>
+        /// <param name="isLocalPath">True if this is a local filesystem path, false if it's a relative repository path</param>
+        public static ValidationResult ValidateTypeSpecDir(string? path, bool isLocalPath = true)
+        {
+            ValidationResult pathTraversalValidation = ValidateDirTraversal(path, "TypeSpec path");
+            if (!pathTraversalValidation.IsValid)
+            {
+                return pathTraversalValidation;
+            }
+
+            try
+            {
+                if (isLocalPath)
+                {
+                    string fullPath = Path.GetFullPath(pathTraversalValidation.Value);
+                    
+                    if (!Directory.Exists(fullPath))
+                    {
+                        return ValidationResult.Invalid($"TypeSpec directory not found: {fullPath}");
+                    }
+                    
+                    string[] allFiles = Directory.GetFiles(fullPath, "*.*", SearchOption.TopDirectoryOnly);
+                    bool hasTypeSpecFiles = allFiles
+                        .Any(file => TypeSpecFileRegex.IsMatch(file));
+
+                    if (!hasTypeSpecFiles)
+                    {
+                        return ValidationResult.Invalid($"No .tsp or .yaml files found in directory: {fullPath}");
+                    }
+                    
+                    if (allFiles.Any(file => !TypeSpecFileRegex.IsMatch(file)))
+                    {
+                        string[] nonTypeSpecFiles = allFiles
+                            .Where(file => !TypeSpecFileRegex.IsMatch(file))
+                            .ToArray();
```

Comment:
```suggestion
```

No need (or value) in doing an array allocation here.   The `string.Join` that you're calling has an `IEnumerable<string>` overload and will walk the set without you having to materialize it first.